apps/federatedfilesharing/lib/Ocm/Permissions.php

```diff
@@ -77,4 +77,8 @@ public function toOcPermissions($ocmPermissions) {
 
 		return $ocPermissions;
 	}
+
+	public static function isNewPermissionHigher(int $existingPermission, int $newPermission): bool {
+		return ($existingPermission | $newPermission) === $existingPermission;
+	}
```

Comment:
```suggestion
	public static function isNewPermissionLower(int $existingPermission, int $newPermission): bool {
		return ($existingPermission | $newPermission) === $existingPermission;
	}
```

IMO this function returns `true` if the new permission set is lower (or equal) to the current permissions. So I suggest changing the function name.

If it matters, this could also check that `$newPermission !== $existingPermission` so that it only returns `true` if the new permissions are strictly lower.

Reply:
See #40829